testsuite/forge-cli/src/main.rs

```diff
@@ -261,7 +261,7 @@ pub fn run_forge<F: Factory>(
         }
         Err(e) => {
             eprintln!("Failed to run tests:\n{}", e);
-            process::exit(101); // Exit with a non-zero exit code if tests failed
+            Err(e)
```

Comment:
Do we know why they originally wanted to kill the process? i.e., my concern is that there might be things left behind (still running) with this change?

Reply:
Not sure why we wanted to originally kill the process. Re your concern, it's the opposite actually: forge swarm relies on some custom destructors by impl `Drop`, which in the k8s would put the cluster in an idle state for cost savings. Previously, if tests failed in the right spot, the cluster would be left running at max capacity until another test was scheduled

Reply:
My gut feeling is that there may be concerns about not cleanly exiting all the threads. What motivated this change?

Reply:
Forge wasn't calling the destructors to clean up the k8s clusters, resulting in lost $$$. Returning the error here rather than process exiting fixes that.

Reply:
Oh, interesting, this is what I was referring to with "What makes you say that? Nothing was changed in the CLI for this PR other than renaming of one function"

Reply:
oh :p thought you were talking about the aptos CLI. yea this is a necessary change. Forge k8s and local backend both have custom destructors, and I just verified that the exit 101 isn't used anywhere in Forge